sdk/azcore/credential.go

```diff
@@ -50,4 +50,6 @@ type AccessToken struct {
 type TokenRequestOptions struct {
 	// Scopes contains the list of permission scopes required for the token.
 	Scopes []string
+	// TenantID contains the tenant ID to use in a multi-tenant authentication scenario.
+	TenantID string
```

Comment:
This isn't used at present, is that intentional?

Also, should the comment be updated to include a note that this replaces the tenant ID used to create the credential when set?

Reply:
Yes, adding this first to tag this version and reference it in the multi-tenant work. If not the multi-tenant PR would fail tests since this field doesnt exist on the options. I'll update the comment.

Reply:
But isn't this used in the bearer token policy which is in `azidentity`?

Reply:
I dont follow....this field didnt exist before now. 🤔

Reply:
For what it's worth this will also be used when adding support for multi-tenant auth challenges used in services like keyvault

Reply:
Here it is in the .NET implementation: https://github.com/Azure/azure-sdk-for-net/blob/42d70236d406451f0323fa4de31ee1dab403d2aa/sdk/core/Azure.Core/src/TokenRequestContext.cs#L50

Reply:
Resolving this since we discussed this offline.